src/lib/socket.c

```diff
@@ -2362,23 +2371,22 @@ static long bio_ctrl(BIO *bio, int cmd, long udata_i, void *udata_p) {
 } /* bio_ctrl() */
 
 static int bio_create(BIO *bio) {
-	bio->init = 0;
-	bio->shutdown = 0;
-	bio->num = 0;
-	bio->ptr = NULL;
+	BIO_set_init(bio, 0);
+	BIO_set_shutdown(bio, 0);
+	BIO_set_data(bio, NULL);
 
 	return 1;
 } /* bio_create() */
 
 static int bio_destroy(BIO *bio) {
-	bio->init = 0;
-	bio->shutdown = 0;
-	bio->num = 0;
-	bio->ptr = NULL;
+	BIO_set_init(bio, 0);
+	BIO_set_shutdown(bio, 0);
+	BIO_set_data(bio, NULL);
 
```

Comment:
libressl defines OPENSSL_VERSIO_NUMBER as 0x20000000L. I can add the HAVE_foo stuff, though.

But, also, you can only depend on OpenSSL 1.1 having threads built-in if `defined OPENSSL_THREADS`.